### DISCUSSION TITLE
folds and inference

### DISCUSSION THREAD

**Discussion by ebbell** [MIC-DKFZ/nnUNet#1012] [Q&A] [answered]
First of many thanks for providing the nnU-net for us all to use.

I have one question regarding the five-fold cross-validation: From your paper and documentation it seems that some configuration of the post-processing is based on results in the cross-validation and also in deciding on the best architecture (2D vs 3D vs 3D cascade). That definitely makes a lot of sense, however when it comes to inference - what do you actually do? Say the basic 3D net was the best performing - which of the five trained 3D folds do you actually use? Or, do you have some fancy way to combine the five folds into one model? 

Many thanks in advance for your reply and sorry if I missed any existing description of this

**Comment by FabianIsensee** [comment, marked as answer]
Hi there,
we always use the 5 models from the cross-validation as an ensemble. Ensembling is implemented by averaging the softmax predictions
Best,
Fabian
